pkg/scalers/azure_eventhub_scaler.go

```diff
@@ -0,0 +1,256 @@
+		return &AzureEventHubScaler{}, fmt.Errorf("unable to get storage credentials: %s", err)
+	}
+	eventHubScaler.storageCredentials = cred
+
+	hub, err := GetEventHubClient(parsedMetadata.eventHubConnection)
+	if err != nil {
+		return &AzureEventHubScaler{}, fmt.Errorf("unable to get eventhub client: %s", err)
+	}
+	eventHubScaler.client = hub
+
+	leaserCheckpointer, err := GetLeaserCheckpointer(parsedMetadata.storageConnection, parsedMetadata.storageContainerName)
+	if err != nil {
+		return &AzureEventHubScaler{}, fmt.Errorf("unable to get leaser/checkpointer: %s", err)
+	}
+	eventHubScaler.leaserCheckpointer = leaserCheckpointer
+
+	return &eventHubScaler, nil
+}
+
+// ParseAzureEventHubMetadata parses metadata
+func ParseAzureEventHubMetadata(metadata, resolvedEnv map[string]string) (*EventHubMetadata, error) {
+	meta := EventHubMetadata{}
+	meta.threshold = defaultEventHubMessageThreshold
+
+	if val, ok := metadata[thresholdMetricName]; ok {
+		threshold, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			log.Errorf("Error parsing azure eventhub metadata %s: %s", thresholdMetricName, err)
+		} else {
+			meta.threshold = threshold
+		}
+	}
+
+	storageConnectionSetting := defaultStorageConnectionString
+	if val, ok := metadata["storageConnection"]; ok && val != "" {
+		storageConnectionSetting = val
+	}
+
+	if val, ok := resolvedEnv[storageConnectionSetting]; ok {
+		meta.storageConnection = val
+	} else {
+		return nil, fmt.Errorf("no storage connection setting given")
+	}
+
+	eventHubConnectionSetting := defaultEventHubConnectionString
+	if val, ok := metadata["eventHubConnection"]; ok && val != "" {
+		eventHubConnectionSetting = val
+	}
+
+	if val, ok := resolvedEnv[eventHubConnectionSetting]; ok {
+		meta.eventHubConnection = val
+	} else {
+		return nil, fmt.Errorf("no event hub connection setting given")
+	}
+
+	if val, ok := metadata["eventHubName"]; ok {
+		meta.eventHubName = val
+	} else {
+		return nil, fmt.Errorf("no eventHubName given")
+	}
+
+	if val, ok := metadata["storageContainerName"]; ok {
+		meta.storageContainerName = val
+	} else {
+		return nil, fmt.Errorf("no storageContainerName given")
+	}
+
+	return &meta, nil
+}
+
+//GetUnprocessedEventCountInPartition gets number of unprocessed events in a given partition
+func (scaler *AzureEventHubScaler) GetUnprocessedEventCountInPartition(ctx context.Context, partitionID string) (newEventCount int64, err error) {
+	partitionInfo, err := scaler.client.GetPartitionInformation(ctx, partitionID)
+	if err != nil {
+		return -1, fmt.Errorf("unable to get partition info: %s", err)
+	}
+
+	lease, err := GetLeaseFromBlobStorage(ctx, partitionID, scaler.metadata.storageConnection, scaler.metadata.storageContainerName)
+	if err != nil {
+		return -1, fmt.Errorf("unable to get lease from storage: %s", err)
+	}
+
+	checkpoint := lease.Checkpoint
+
+	unprocessedEventCountInPartition := int64(0)
+
+	if checkpoint.SequenceNumber != partitionInfo.LastSequenceNumber {
```

Comment:
Can you make sure to include the event hub checkpoint circular buffer checks the Event hub trigger has?